### PR TITLE
NW6 | Rabia Avci | Full-Stack-Project | Week-3 | Update videos endpoint

### PR DESCRIPTION
**Pull request by RbAvci**
Completed these tasks for this ticket:

- Created an endpoint to update video ratings.
- Configured the endpoint to accept the appropriate HTTP method.
- Defined a suitable URL for the endpoint.
- Implemented functionality for the endpoint to update video ratings in the database.

### CODE REVIEW

File: db/initdb.sql [hunk 1]
```diff
@@ -5,17 +5,18 @@ DROP TABLE IF EXISTS videos CASCADE;
...
-INSERT INTO videos (title,src) VALUES ('Mac & Cheese | Basics with Babish','https://www.youtube.com/watch?v=FUeyrEN14Rk');
-INSERT INTO videos (title,src) VALUES ('Videos for Cats to Watch - 8 Hour Bird Bonanza','https://www.youtube.com/watch?v=xbs7FT7dXYc');
-INSERT INTO videos (title,src) VALUES ('The Complete London 2012 Opening Ceremony | London 2012 Olympic Games','https://www.youtube.com/watch?v=4As0e4de-rI');
-INSERT INTO videos (title,src) VALUES ('Learn Unity - Beginner''s Game Development Course','https://www.youtube.com/watch?v=gB1F9G0JXOo');
-INSERT INTO videos (title,src) VALUES ('Cracking Enigma in 2021 - Computerphile','https://www.youtube.com/watch?v=RzWB5jL5RX0');
-INSERT INTO videos (title,src) VALUES ('Coding Adventure: Chess AI','https://www.youtube.com/watch?v=U4ogK0MIzqk');
-INSERT INTO videos (title,src) VALUES ('Coding Adventure: Ant and Slime Simulations','https://www.youtube.com/watch?v=X-iSQQgOd1A');
-INSERT INTO videos (title,src) VALUES ('Why the Tour de France is so brutal','https://www.youtube.com/watch?v=ZacOS8NBK6U');
+INSERT INTO videos (title,src,rating) VALUES ('Never Gonna Give You Up','https://www.youtube.com/watch?v=dQw4w9WgXcQ', 5);
```
Comment (Ara225) [inline db/initdb.sql:13]: This is quite an interesting choice to me. Without  the users' input, you can't know what their rating is. Personally, I would always use null or -1 as a default value here.

File: server/api.js [hunk 2]
```diff
@@ -42,4 +42,25 @@ router.delete("/videos/:id", async (req, res) => {
 	}
 });
 
+router.put("/videos/:id/rating", async (req, res) => {
+	const videoId = req.params.id;
+	const { rating } = req.body;
+
+	try {
+		const result = await db.query(`SELECT * FROM videos WHERE id=${videoId}`);
```
Comment (Ara225) [inline server/api.js:50]: As mentioned last time, this way of  composing SQL isn't very secure - talk to Zeliha about the way she does it which is correct

File: server/api.js [hunk 3]
```diff
@@ -42,4 +42,25 @@ router.delete("/videos/:id", async (req, res) => {
...
+			return res.status(404).json({ error: "Video not found" });
+		}
+
+		await db.query(`UPDATE videos SET rating=${rating} WHERE id=${videoId}`);
+
+		res
+			.status(200)
+			.json({ success: true, message: "Video rating successfully" });
+	} catch (error) {
+		res.status(500).json({ success: false, error: "Internal server error" });
```
Comment (Ara225) [inline server/api.js:62]: It's good that you're using an appropriate error code, but I would like  to see you being  more specific here, yes an internal server error happened but why?

File: server/api.js [hunk 4]
```diff
@@ -42,4 +42,25 @@ router.delete("/videos/:id", async (req, res) => {
...
 
+router.put("/videos/:id/rating", async (req, res) => {
+	const videoId = req.params.id;
+	const { rating } = req.body;
+
+	try {
+		const result = await db.query(`SELECT * FROM videos WHERE id=${videoId}`);
+		let video = result.rows[0];
+		if (!video) {
+			return res.status(404).json({ error: "Video not found" });
```
Comment (Ara225) [inline server/api.js:53]: 👍

**Review by Ara225 — COMMENTED** [review]
Some feedback on the implementation details, but all in all, it works 👍

**Review by zelihapala — APPROVED** [review]

File: db/initdb.sql (hunk 1 above)
Comment (RbAvci) [reply db/initdb.sql]: Thanks, Anna, default rate is set to NULL. I will update the front-end with 'not rated' when the value is null.

File: server/api.js (hunk 2 above)
Comment (RbAvci) [reply server/api.js]: done it!

**Pull request merged**